### PR TITLE
Add a simple parser

### PR DESCRIPTION
**Pull request by jihoonson**
This patch consists of a lexer and a parser.
The lexer parses an input string using regex, and returns a list of tokens.
The parser creates an AST from the parsed tokens.

This patch does not contain the features which are basically needed for test.
Here is a list of not-implemented features.
* Currently, the output of every expression is fixed as I64.
* Literal strings, fields, function calls cannot be recognized yet.

For your convenience, I'd like to share the current status. In addition, if you want, I'll add remaining features in the subsequent PRs.

[Review on Reviewable](https://reviewable.io/reviews/hyunsik/tokamak/8)

### CODE REVIEW

**Comment by jihoonson** [comment]
Of course, any comments on Rust code are welcome.

File: src/common/types.rs [hunk 1]
```diff
@@ -227,30 +227,30 @@ impl Ty
...
 {
   match size {
-    8  => U8,
+    1  => U8,
     _  => panic!("unsupported integer type: i{}", size)
   }
 }
 
-pub fn i(size: u32) -> &'static Ty
+pub fn i(size: usize) -> &'static Ty
```
Comment (hyunsik) [inline src/common/types.rs]: It was intended to be equivalent to llvm types which are globally used in this project. So, bit size is my intention.

File: src/common/types.rs [hunk 2]
```diff
@@ -227,30 +227,30 @@ impl Ty
...
   }
 }
 
-pub fn f(size: u32) -> &'static Ty
+pub fn f(size: usize) -> &'static Ty
 {
   match size {
-    32 => F32,
-    64 => F64,
+    4 => F32,
```
Comment (hyunsik) [inline src/common/types.rs]: It was intended to be equivalent to llvm types which are globally used in this project. So, bit size is my intention.

File: src/exec/processor.rs [hunk 3]
```diff
@@ -103,8 +103,8 @@ impl MapCompiler {
 
     let const_exprs = izip!(0..exprs.len(), exprs).filter(|e| match *e.1.kind() {ExprKind::Const(_) => true, _ => false});
     let nonconst_exprs = izip!(0..exprs.len(), exprs).filter(|e| match *e.1.kind() {ExprKind::Const(_) => false, _ => true});
-    
-    let schema_map = schema.to_map();    
+
```
Comment (hyunsik) [inline src/exec/processor.rs]: The changes for spaces may be affected by some editor configurations. We may need to decide some coding standard rules and editor configurations.

File: src/exec/processor.rs [hunk 4]
```diff
@@ -406,11 +406,11 @@ impl<'a> ExprCompiler<'a> {
     let builder = &self.builder;
 
     self.stack.push(match *op {
-      ArithmOp::Plus => {builder.create_add(&lhs_val, &rhs_val)}
+      ArithmOp::Add => {builder.create_add(&lhs_val, &rhs_val)}
```
Comment (hyunsik) [inline src/exec/processor.rs]: It need to be a separate issue because it seems to be not related to this issue.

File: src/common/types.rs (hunk 1 above)
Comment (jihoonson) [inline src/common/types.rs]: Ok. If so, how about changing the ```size_of()``` function to return the number of bits in ```u32``` type?

File: src/exec/processor.rs (hunk 3 above)
Comment (jihoonson) [inline src/exec/processor.rs]: Right. These spaces were automatically removed by Atom. 
In addition, I'm quite confused on the location of the left parenthesis. In some lines, the left parentheses locate at the end of the line, while others locate at the start of the new line.
Do you have any preference on coding convention?

File: src/exec/processor.rs (hunk 4 above)
Comment (jihoonson) [inline src/exec/processor.rs]: Ok. I'll revert.

**Comment by hyunsik** [comment]
merged. Thank you for your contribution.

**Comment by jihoonson** [comment]
Thanks for review and merging patch!
